core/src/main/scala/za/co/absa/spline/harvester/converter/ReflectiveExtractor.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.converter
+
+import org.apache.spark.internal.Logging
+import za.co.absa.commons.reflect.ReflectionUtils
+
+import java.lang.reflect.{Field, Method}
+
+object ReflectiveExtractor extends Logging{
+
+  def extractProperties(obj: AnyRef): Map[String, _] =
+    try ReflectionUtils.extractProperties(obj)
+    catch {
+      // a workaround for Scala bug #12190
+      // catching everything because the bug causes various errors and exceptions
+      case e =>
```

Comment:
Note that this will catch everything including fatal errors and exceptions that require special handling. To filter them out, `case NonFatal(e)` or `Try` should be used instead.

Reply:
I thought NonFatal filters out all exceptions. Including the ones cause by the Scala bug. But I looked it up, and it's only a few serious exceptions, and the rest goes through. In that case, NonFatal can be used here.